framework/src/headers/ip/v6/icmp/neighbor.rs

```diff
@@ -0,0 +1,118 @@
+use super::{IcmpMessageType, Icmpv6Header};
+use headers::{CalcChecksums, EndOffset, Ipv6VarHeader};
+use std::default::Default;
+use std::fmt;
+use std::marker::PhantomData;
+use std::net::Ipv6Addr;
+
+#[derive(Debug)]
+#[repr(C)]
+pub struct Icmpv6Neighbor<T>
+where
+    T: Ipv6VarHeader,
+{
+    pub icmp: Icmpv6Header<T>,
+    pub reserved_flags: u32,
+    pub target_addr: Ipv6Addr,
+    pub _parent: PhantomData<T>,
+}
+
+impl<T> Default for Icmpv6Neighbor<T>
+where
+    T: Ipv6VarHeader,
+{
+    fn default() -> Icmpv6Neighbor<T> {
+        Icmpv6Neighbor {
+            icmp: Icmpv6Header {
+                ..Default::default()
+            },
+
+            reserved_flags: 0,
+            target_addr: Ipv6Addr::UNSPECIFIED,
+            _parent: PhantomData,
+        }
+    }
+}
+
+impl<T> fmt::Display for Icmpv6Neighbor<T>
+where
+    T: Ipv6VarHeader,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "msg_type: {} code: {} checksum: {}, reserved_flags {}, target_addr {}",
```

Comment:
is there an advantage of doing this simple formatting / `Display` over just `#[derive(Debug)]`?